runhouse/logger.py

```diff
@@ -97,35 +100,82 @@ def formatTime(self, record, datefmt=None):
             return dt.isoformat(timespec="milliseconds")
 
 
-LOGGING_CONFIG = {
-    "version": 1,
-    "disable_existing_loggers": True,
-    "formatters": {
-        "utc_formatter": {
-            "()": UTCFormatter,
-            "format": "%(levelname)s | %(asctime)s | %(message)s",
-            "datefmt": "%Y-%m-%d %H:%M:%S.%f",
-        },
-    },
-    "handlers": {
-        "default": {
-            "level": "INFO",
-            "formatter": "utc_formatter",
-            "class": "logging.StreamHandler",
-            "stream": "ext://sys.stderr",  # Default is stderr
+def get_logger(name: str = __name__, log_level: Union[int, str] = logging.INFO):
+    level_name = (
+        logging.getLevelName(log_level) if isinstance(log_level, int) else log_level
+    )
+    LOGGING_CONFIG = {
+        "version": 1,
+        "disable_existing_loggers": True,
+        "formatters": {
+            "utc_formatter": {
+                "()": UTCFormatter,
+                "format": "%(levelname)s | %(asctime)s | %(message)s",
+                "datefmt": "%Y-%m-%d %H:%M:%S.%f",
+            },
         },
-    },
-    "loggers": {
-        "": {  # root logger
-            "handlers": ["default"],
-            "level": "INFO",
-            "propagate": False,
+        "handlers": {
+            "default": {
+                "level": level_name,
+                "formatter": "utc_formatter",
+                "class": "logging.StreamHandler",
+                "stream": "ext://sys.stderr",  # Default is stderr
+            },
         },
-        "my.packg": {"handlers": ["default"], "level": "INFO", "propagate": False},
-        "__main__": {  # if __name__ == '__main__'
-            "handlers": ["default"],
-            "level": "DEBUG",
-            "propagate": False,
+        "loggers": {
+            "": {  # root logger
+                "handlers": ["default"],
+                "level": level_name,
+                "propagate": False,
+            },
+            "my.packg": {
+                "handlers": ["default"],
+                "level": level_name,
+                "propagate": False,
+            },
+            "__main__": {  # if __name__ == '__main__'
+                "handlers": ["default"],
+                "level": level_name,
+                "propagate": False,
+            },
         },
-    },
-}
+    }
+    logging.config.dictConfig(LOGGING_CONFIG)
+    logger = logging.getLogger(name=name)
+    return logger
+
+
+logger = get_logger(name=__name__, log_level=DEFAULT_LOG_LEVEL)
```

Comment:
what's this used for?

Reply:
actually this might give us a cleaner way of importing the logger everywhere else - i think we can then do: 

```
from runhouse.logger import logger
logger.info("Using the default logger instance")```